spec/default_facts.yml

```diff
@@ -2,7 +2,8 @@
 #
 # Facts specified here will override the values provided by rspec-puppet-facts.
 ---
-ipaddress: "172.16.254.254"
-ipaddress6: "FE80:0000:0000:0000:AAAA:AAAA:AAAA"
+networking:
```

Comment:
This is a bad idea. Looking at the code:
https://github.com/puppetlabs/puppetlabs-stdlib/blob/45a90b8d65c5600ad95f0ca394024f9c0851962d/spec/spec_helper.rb#L19-L32

That's a simple merge, not a deep merge. That means you'll break the networking fact since MANY facts will no longer be present.

Reply:
Good spot, this explains some of the failures we have seen.
We are going to have to opt with the `deep_merge` like you mentioned, I'll work to get this rolled out.

Reply:
Perhaps a good excuse to do it natively in rspec-puppet-facts? https://github.com/voxpupuli/voxpupuli-test#fact-handling is how we've done it in voxpupuli-test now ([implementation](https://github.com/voxpupuli/voxpupuli-test/blob/master/lib/voxpupuli/test/facts.rb)), but there's really no good reason (other than being able to roll it out quicker) that it should live in voxpupuli-test.

That said, I still think it would be ideal if you could avoid `default_facts.yml` altogether. A file you always sync to many repositories is something I'd consider an anti-pattern.

Reply:
Totally agree with what you're saying..
I'll raise this with the team to triage and see if we can get it assigned out.